lib/components/scanner/scanner_widget.dart

```diff
@@ -1,92 +0,0 @@
-import 'package:flutter/material.dart';
-import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:qr_code_scanner/qr_code_scanner.dart';
-import 'package:seeds/components/scanner/interactor/scanner_bloc.dart';
-import 'package:seeds/components/scanner/interactor/viewmodels/scanner_events.dart';
-import 'package:seeds/components/scanner/interactor/viewmodels/scanner_state.dart';
-import 'package:seeds/components/scanner/seeds_qr_code_scanner_widget.dart';
-import 'package:seeds/utils/string_extension.dart';
-
-/// Scanner SCREEN
-class ScannerWidget extends StatefulWidget {
-  final ScannerBloc _scannerBloc = ScannerBloc();
-  final ValueSetter<String> resultCallBack;
-
-  ScannerWidget({Key? key, required this.resultCallBack}) : super(key: key);
-
-  void scan() => _scannerBloc.add(Scan());
-
-  void showLoading() => _scannerBloc.add(ShowLoading());
-
-  void stop() => _scannerBloc.add(Stop());
-
-  @override
-  _ScannerWidgetState createState() => _ScannerWidgetState();
-}
-
-class _ScannerWidgetState extends State<ScannerWidget> {
-  final GlobalKey _qrKey = GlobalKey(debugLabel: 'QR');
-  late QRViewController _controller;
-  bool _handledQrCode = false;
-
-  @override
-  Widget build(BuildContext context) {
-    return BlocProvider(
-      create: (_) => widget._scannerBloc,
-      child: BlocBuilder<ScannerBloc, ScannerState>(
-        builder: (context, state) {
-          if (state.scanStatus is Stop) {
-            _controller.dispose();
-            const SizedBox.shrink();
-          }
```

Comment:
I'm not sure if this worked before because:

1. This is a bad casting they are different types, one is an enum and the other an event.
2. Second there is a widget without a return.
3. I can't find the reason to dispose the controller there or even why we want to stop it, I mean the scanner widget is unmounted when the parent widget does too.